### ISSUE TITLE
net/wireguard: "setsockopt IP_MULTICAST_IF: Operation not supported"

### ISSUE THREAD

**Issue by oli-h** [opnsense/plugins#2314]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[x] I have searched the existing issues and I'm convinced that mine is new.

[x] The title contains the plugin to which this issue belongs

**Describe the bug**
Sinde OpnSense 21.1.4 I see this problem
```
root@OPNsense:~ # udpbroadcastrelay --id 1 --dev igb0 --dev wg0 --port 1900 --multicast 239.255.255.250 -d
ID set to 1
Port set to 1900
ID: 1 (DSCP: 1, ToS: 0x04), Port 1900
igb0: 1 / 192.168.1.1 / 192.168.1.255
wg0: 11 / 192.168.4.1 / 192.168.4.255
setsockopt IP_MULTICAST_IF: Operation not supported
found 2 interfaces total
IP_ADD_MEMBERSHIP:		192.168.1.1 239.255.255.250
IP_ADD_MEMBERSHIP:		192.168.4.1 239.255.255.250
IP_ADD_MEMBERSHIP on rcv: Can't assign requested address
```
I.e.: Using `net/udpbroadcastrelay` on a wireguard-interface (here `wg0`) leades to **setsockopt IP_MULTICAST_IF: Operation not supported**.
Note that I _think_ the problem is in WireGuard (and not in net/udpbroadcastrelay) as WireGuard was updated to 1.5 in OpnSence 21.1.4 whereas net/udpbroadcastrelay is still same version. Problem did not occur before 21.1.4.

**To Reproduce**
1. Setup a (simple - straightforward) "VPN WireGuard"
2. Setup "Service UDP Broadcast Relay" with Port 1900, Multicast-Address 239.255.255.250 and add your LAN and the wireguard-interface (typically wg0)
3. You see that UDP Broadcast Relay does not start
4. Crosscheck with various interface-combinations in "UDP Broadcast Relay"-Setup: When wg0 is chosen --> problem. When not --> "UDP Broadcast Relay" runs well

**Expected behavior**
WireGuard-Interface should support socket option `IP_MULTICAST_IF`

**Relevant log files**
see description

**Environment**
OPNsense 21.1.4-amd64
FreeBSD 12.1-RELEASE-p15-HBSD
OpenSSL 1.1.1k 25 Mar 2021
CPU type AMD GX-412TC SOC (4 cores)

**Comment by mimugmail** [comment]
It seems 21.1.4 ship the newest version of WireGuard where interface Type was changed by upstream :( https://forum.opnsense.org/index.php?topic=22378.msg106352#msg106352

Might also be the cause of couple of other probs

**Comment by oli-h** [comment]
Updated to OpnSense 21.1.6: Problem seems to be solved.
In my specific case: The plugin "udpbroadcastrelay" now works together with the WireGuard-Interface

**Comment by mimugmail** [comment]
Great, would you mind closing the issue? Thx :)

**Comment by oli-h** [comment]
Solved with 21.1.6
